### DISCUSSION TITLE
How to migrate from bolt db to postgres?

### DISCUSSION THREAD

**Discussion by dannydeezy** [lightningnetwork/lnd#6508] [General]
Is there a doc recommending how to migrate an existing node from using the bolt db backend to postgres? I'd like to try migrating because I assume using postgres would mean Lnd won't take several minutes to start up like it does with bolt?

**Comment by thinkmassive** [comment]
The migration tool is tracked in #5561 

Please note the latest comment, regarding whether this is ready for production use:
> No, this hasn't been tested sufficiently and isn't recommended to be used (which is also the main reason for this not to be merged yet).

Converting this issue to a Discussion, as I'm sure others will have the same question going forward.

**Comment by thinkmassive** [comment]
To answer your last question @dannydeezy, postgres does provide a significant improvement to LND startup duration by eliminating the need for bolt to load the entire db from storage. Postgres can also do garbage collection ("vacuum" in postgres terminology) during runtime, which is an improvement compared to the compaction that's occasionally required during startup when using bolt.

Using a remote database also provides an avenue to additional improvements such as fault tolerance via [leader election](https://github.com/lightningnetwork/lnd/blob/master/docs/leader_election.md), and other load balancing solutions.
